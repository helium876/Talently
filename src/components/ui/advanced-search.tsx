'use client'

import { useState } from 'react'
import { Search } from 'lucide-react'
import { Input } from './input'

interface AdvancedSearchProps {
  onSearch: (query: string) => void
  placeholder?: string
}

export function AdvancedSearch({ onSearch, placeholder = 'Search...' }: AdvancedSearchProps) {
  const [query, setQuery] = useState('')

  const handleSearch = (value: string) => {
    setQuery(value)
    onSearch(value)
  }

  return (
    <div className="relative">
      <div className="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none">
        <Search className="h-4 w-4 text-gray-400" />
      </div>
      <Input
        type="search"
        placeholder={placeholder}
        value={query}
        onChange={(e) => handleSearch(e.target.value)}
        className="pl-10 w-full md:w-[300px]"
      />
    </div>
  )
} 