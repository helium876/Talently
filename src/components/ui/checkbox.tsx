'use client'

import * as React from 'react'
import { cn } from '@/lib/utils'
import { Check } from 'lucide-react'

export interface CheckboxProps
  extends React.InputHTMLAttributes<HTMLInputElement> {
  onCheckedChange?: (checked: boolean) => void
}

const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
  ({ className, onCheckedChange, ...props }, ref) => {
    return (
      <div className="relative inline-flex">
        <input
          type="checkbox"
          className="peer sr-only"
          ref={ref}
          onChange={e => onCheckedChange?.(e.target.checked)}
          {...props}
        />
        <div
          className={cn(
            'h-4 w-4 shrink-0 rounded-sm border border-gray-300',
            'ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-blue-500 focus-visible:ring-offset-2',
            'peer-disabled:cursor-not-allowed peer-disabled:opacity-50',
            'peer-checked:bg-blue-600 peer-checked:border-blue-600',
            className
          )}
        >
          <Check className="h-3 w-3 text-white hidden peer-checked:block absolute top-0.5 left-0.5" />
        </div>
      </div>
    )
  }
)
Checkbox.displayName = 'Checkbox'

export { Checkbox } 