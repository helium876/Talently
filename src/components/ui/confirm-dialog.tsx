import React from 'react'
import {
  AlertDialog,
  AlertDialogAction,
  AlertDialogCancel,
  AlertDialogContent,
  AlertDialogDescription,
  AlertDialogFooter,
  AlertDialogHeader,
  AlertDialogTitle,
} from './alert-dialog'
import { Button } from './button'

interface ConfirmDialogProps {
  title: string
  description: string
  open: boolean
  onConfirm: () => void
  onCancel: () => void
  confirmText?: string
  cancelText?: string
  variant?: 'default' | 'destructive'
}

export function ConfirmDialog({
  title,
  description,
  open,
  onConfirm,
  onCancel,
  confirmText = 'Confirm',
  cancelText = 'Cancel',
  variant = 'default'
}: ConfirmDialogProps) {
  return (
    <AlertDialog open={open}>
      <AlertDialogContent>
        <AlertDialogHeader>
          <AlertDialogTitle>{title}</AlertDialogTitle>
          <AlertDialogDescription>{description}</AlertDialogDescription>
        </AlertDialogHeader>
        <AlertDialogFooter>
          <AlertDialogCancel onClick={onCancel}>{cancelText}</AlertDialogCancel>
          <AlertDialogAction asChild>
            <Button variant={variant} onClick={onConfirm}>
              {confirmText}
            </Button>
          </AlertDialogAction>
        </AlertDialogFooter>
      </AlertDialogContent>
    </AlertDialog>
  )
} 