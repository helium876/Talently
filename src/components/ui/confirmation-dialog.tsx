'use client'

import { useState } from 'react'
import { Card } from './card'
import { Button } from './button'
import { AlertTriangle, Loader2 } from 'lucide-react'

interface ConfirmationDialogProps {
  title: string
  message: string
  confirmLabel?: string
  cancelLabel?: string
  isDestructive?: boolean
  isLoading?: boolean
  onConfirm: () => void | Promise<void>
  onCancel: () => void
}

export function ConfirmationDialog({
  title,
  message,
  confirmLabel = 'Confirm',
  cancelLabel = 'Cancel',
  isDestructive = false,
  isLoading = false,
  onConfirm,
  onCancel
}: ConfirmationDialogProps) {
  const [isConfirming, setIsConfirming] = useState(false)

  const handleConfirm = async () => {
    setIsConfirming(true)
    try {
      await onConfirm()
    } finally {
      setIsConfirming(false)
    }
  }

  return (
    <div className="fixed inset-0 z-50 flex items-center justify-center bg-black/50">
      <Card className="w-full max-w-md p-6">
        <div className="flex items-start gap-4">
          <div className="flex-shrink-0">
            <AlertTriangle className={`h-6 w-6 ${isDestructive ? 'text-red-500' : 'text-yellow-500'}`} />
          </div>
          <div className="flex-1">
            <h3 className="text-lg font-semibold text-gray-900">
              {title}
            </h3>
            <p className="mt-2 text-sm text-gray-500">
              {message}
            </p>
            <div className="mt-6 flex justify-end gap-3">
              <Button
                variant="outline"
                onClick={onCancel}
                disabled={isConfirming || isLoading}
              >
                {cancelLabel}
              </Button>
              <Button
                variant={isDestructive ? 'destructive' : 'default'}
                onClick={handleConfirm}
                disabled={isConfirming || isLoading}
              >
                {isConfirming || isLoading ? (
                  <>
                    <Loader2 className="mr-2 h-4 w-4 animate-spin" />
                    Processing...
                  </>
                ) : (
                  confirmLabel
                )}
              </Button>
            </div>
          </div>
        </div>
      </Card>
    </div>
  )
} 