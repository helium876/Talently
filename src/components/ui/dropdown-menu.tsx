'use client'

import * as React from 'react'
import { cn } from '@/lib/utils'

interface DropdownMenuProps {
  children: React.ReactNode
}

interface DropdownMenuTriggerProps {
  children: React.ReactNode
  asChild?: boolean
}

interface DropdownMenuContentProps extends React.HTMLAttributes<HTMLDivElement> {
  children: React.ReactNode
  align?: 'start' | 'end'
}

interface DropdownMenuItemProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  children: React.ReactNode
}

const DropdownMenuContext = React.createContext<{
  open: boolean
  setOpen: (open: boolean) => void
  triggerRef: React.RefObject<HTMLElement>
}>({
  open: false,
  setOpen: () => {},
  triggerRef: React.createRef()
})

export function DropdownMenu({ children }: DropdownMenuProps) {
  const [open, setOpen] = React.useState(false)
  const triggerRef = React.useRef<HTMLElement>(null)

  return (
    <DropdownMenuContext.Provider value={{ open, setOpen, triggerRef }}>
      {children}
    </DropdownMenuContext.Provider>
  )
}

export function DropdownMenuTrigger({ children, asChild }: DropdownMenuTriggerProps) {
  const { open, setOpen, triggerRef } = React.useContext(DropdownMenuContext)

  const handleClick = (e: React.MouseEvent) => {
    e.stopPropagation()
    setOpen(!open)
  }

  if (asChild) {
    return React.cloneElement(children as React.ReactElement, {
      onClick: handleClick,
      ref: triggerRef,
      'aria-expanded': open,
      'aria-haspopup': true
    })
  }

  return (
    <button
      onClick={handleClick}
      ref={triggerRef as React.RefObject<HTMLButtonElement>}
      aria-expanded={open}
      aria-haspopup={true}
    >
      {children}
    </button>
  )
}

export function DropdownMenuContent({ 
  children, 
  className,
  align = 'end',
  ...props 
}: DropdownMenuContentProps) {
  const { open, setOpen, triggerRef } = React.useContext(DropdownMenuContext)
  const contentRef = React.useRef<HTMLDivElement>(null)

  React.useEffect(() => {
    const handleClickOutside = (event: MouseEvent) => {
      if (
        contentRef.current &&
        !contentRef.current.contains(event.target as Node) &&
        triggerRef.current &&
        !triggerRef.current.contains(event.target as Node)
      ) {
        setOpen(false)
      }
    }

    if (open) {
      document.addEventListener('mousedown', handleClickOutside)
      return () => document.removeEventListener('mousedown', handleClickOutside)
    }
  }, [open, setOpen, triggerRef])

  if (!open) return null

  return (
    <div
      ref={contentRef}
      className={cn(
        'absolute z-50 min-w-[8rem] overflow-hidden rounded-md border border-gray-200 bg-white p-1 shadow-md',
        'animate-in fade-in-0 zoom-in-95',
        align === 'end' ? 'right-0' : 'left-0',
        className
      )}
      {...props}
    >
      {children}
    </div>
  )
}

export function DropdownMenuItem({ 
  children,
  className,
  ...props
}: DropdownMenuItemProps) {
  const { setOpen } = React.useContext(DropdownMenuContext)

  return (
    <button
      className={cn(
        'relative flex w-full cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none',
        'hover:bg-gray-100 focus:bg-gray-100',
        'data-[disabled]:pointer-events-none data-[disabled]:opacity-50',
        className
      )}
      onClick={() => setOpen(false)}
      {...props}
    >
      {children}
    </button>
  )
} 