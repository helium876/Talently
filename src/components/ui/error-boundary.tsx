'use client'

import { Component, ErrorInfo, ReactNode } from 'react'

interface Props {
  children?: ReactNode
  fallback?: ReactNode
}

interface State {
  hasError: boolean
  error?: Error
}

export class ErrorBoundary extends Component<Props, State> {
  public state: State = {
    hasError: false
  }

  public static getDerivedStateFromError(error: Error): State {
    return { hasError: true, error }
  }

  public componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    console.error('Uncaught error:', error, errorInfo)
  }

  public render() {
    if (this.state.hasError) {
      return (
        this.props.fallback || (
          <div className="flex min-h-screen items-center justify-center bg-gray-50 px-4 py-12 sm:px-6 lg:px-8">
            <div className="w-full max-w-md space-y-8">
              <div className="rounded-md bg-red-50 p-4">
                <div className="flex">
                  <div className="ml-3">
                    <h3 className="text-sm font-medium text-red-800">
                      {this.state.error?.message || 'Something went wrong'}
                    </h3>
                  </div>
                </div>
              </div>
            </div>
          </div>
        )
      )
    }

    return this.props.children
  }
} 