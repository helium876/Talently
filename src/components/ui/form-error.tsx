'use client'

interface FormErrorProps {
  message: string
}

export function FormError({ message }: FormErrorProps) {
  return (
    <p className="text-sm font-medium text-red-600">
      {message}
    </p>
  )
} 