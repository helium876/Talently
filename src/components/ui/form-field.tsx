'use client'

import { Label } from './label'
import { Input } from './input'
import { FormError } from './form-error'

interface FormFieldProps {
  label: string
  name: string
  type?: string
  placeholder?: string
  required?: boolean
  error?: string
  value?: string
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void
}

export function FormField({
  label,
  name,
  type = 'text',
  placeholder,
  required,
  error,
  value,
  onChange,
}: FormFieldProps) {
  return (
    <div className="space-y-1">
      <Label htmlFor={name} className="text-sm font-medium text-gray-900">
        {label}
        {required && <span className="text-red-500 ml-1">*</span>}
      </Label>
      <Input
        type={type}
        id={name}
        name={name}
        placeholder={placeholder}
        required={required}
        value={value}
        onChange={onChange}
        className="w-full text-gray-900 placeholder:text-gray-500"
      />
      {error && <FormError message={error} />}
    </div>
  )
} 