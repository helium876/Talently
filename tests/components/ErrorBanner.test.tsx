import { jest, describe, test, expect } from '@jest/globals'
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import { ErrorBanner } from '@/components/ui/error-banner'

jest.mock('next/navigation', () => ({
  useSearchParams: () => new URLSearchParams('?error=Test%20error%20message'),
}))

describe('ErrorBanner Component', () => {
  test('renders correctly with error message', () => {
    const { container } = render(<ErrorBanner />)
    expect(container).toMatchSnapshot()
  })

  test('renders nothing when no error present', () => {
    jest.spyOn(require('next/navigation'), 'useSearchParams').mockImplementation(
      () => new URLSearchParams('')
    )
    const { container } = render(<ErrorBanner />)
    expect(container.firstChild).toBeNull()
  })

  test('decodes URI encoded error message', () => {
    jest.spyOn(require('next/navigation'), 'useSearchParams').mockImplementation(
      () => new URLSearchParams('?error=Test%20error%20message')
    )
    const { getByText } = render(<ErrorBanner />)
    expect(getByText('Test error message')).toBeInTheDocument()
  })

  test('matches error banner styling', () => {
    const { container } = render(<ErrorBanner />)
    expect(container.firstChild).toHaveClass('rounded-md', 'bg-red-50', 'p-4', 'mb-6')
  })

  test('includes error icon', () => {
    const { container } = render(<ErrorBanner />)
    const icon = container.querySelector('svg')
    expect(icon).toHaveClass('h-5', 'w-5', 'text-red-400')
  })
}) 