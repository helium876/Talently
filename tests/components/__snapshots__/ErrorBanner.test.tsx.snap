// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBanner Component renders correctly with error message 1`] = `
<div>
  <div
    class="rounded-md bg-red-50 p-4 mb-6"
  >
    <div
      class="flex"
    >
      <div
        class="flex-shrink-0"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-circle-x h-5 w-5 text-red-400"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <path
            d="m15 9-6 6"
          />
          <path
            d="m9 9 6 6"
          />
        </svg>
      </div>
      <div
        class="ml-3"
      >
        <h3
          class="text-sm font-medium text-red-800"
        >
          Error
        </h3>
        <div
          class="mt-2 text-sm text-red-700"
        >
          <p>
            Test error message
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
